homeworks/roman.buzukov_BuzRom/3-async/js/script.js

```diff
@@ -0,0 +1,86 @@
+const containerEl = document.querySelector('[data-container]');
+const listEl = document.querySelector('[data-list]');
+const templateEl = document.querySelector('[data-post-template]');
+const sortEl = document.querySelector('[data-sort]');
+const filterEl = document.querySelector('[data-filter]');
+const remove = document.getElementsByClassName('post__delete-img');
+const data = [];
+let posts = data;
+
+const deletePost = (e) => {
+    if (e.target) {
+        const post = e.target.closest('[data-post]');
+        post.setAttribute('style', 'visibility: hidden');
+        containerEl.classList.remove('loaded');
+        setTimeout(() => {
+            fetch(`https://jsonplaceholder.typicode.com/posts/${post.id}`, {
+                method: 'DELETE',
+            })
+                .then((response) => response.json())
+                .then(() => {
+                    alert('Post deleted');
+                    posts = posts.filter((item) => +post.id !== item.id);
+                    post.remove();
+                })
+                .catch(() => {
+                    alert('Something went wrong');
+                    post.setAttribute('style', 'visibility: visible');
+                });
+            containerEl.classList.add('loaded');
+        }, 1000);
+    }
+};
+
+const addDeleteListener = () => {
```

Comment:
The last thing:
Could you please pass `deleteEl` here as an argument? I don't like that you're searching for all elements and set listeners for all of them. In this example, it works, but globally no one does it like this.

This element you can receive when you're adding each item to DOM (renderPosts).

Reply:
Many thanks for your reply!
I tried to put <code>remove</code> variable as argument, but linter is giving error. That’s why variable passed in function directly.
Also, recently I observe interesting major bug in this method to put listener on remove btn. The listener not working on last element. It doesn't matter sorted list or filtered - the last element always out of handle. Can't explain why it’s happening. I’m searching for solution, but thinking to revert to previous version which was described in your previous comment in first recommendation.

Reply:
Oh I understood! You meen this:

const addDeleteListener = (deleteEl) => {
    Array.from(deleteEl).forEach((item) => {
        item.addEventListener('click', deletePost);
    });
};

Reply:
Done!
The bug is fixed. It was so simple 😄.

Reply:
Great! But I have one additional point. I opened a new comment)